### ISSUE TITLE
about flow shape

### ISSUE THREAD

**Issue by jsckdon** [LJY-XCX/RFTrans#3]
When I predict the refractive flow, the channels match at the beginning, but after training for one epoch, the flow and output both become 2, causing a channel mismatch. Is this normal?
<img width="2006" height="350" alt="Image" src="https://github.com/user-attachments/assets/92ff82e8-f7b2-4c91-b354-0126d279db2e" />

**Comment by LJY-XCX** [comment]
Hi Jessica,

I don't think this is expected. For RFTrans training, the output refractive flow should always have 2 channels (dx, dy) and the output surface normal should have 3 channels. Could you provide more detail about the training code?

**Comment by jsckdon** [comment]
Thank you very much for your reply. I’m really interested in your work.
I only ran the refraction flow generation part of the code, not the full end-to-end training. After I adjusted the number of channels to three, the training could run normally. This is the modified version of my utils.py.

<img width="1150" height="916" alt="Image" src="https://github.com/user-attachments/assets/9fd663d1-3a35-4a37-b7dc-ff37b05ccf45" />
I only trained for 100 epochs, and the results seem to be improving. I think training for 500 epochs might lead to better performance.

<img width="2058" height="516" alt="Image" src="https://github.com/user-attachments/assets/21db6133-bc9f-4e92-a857-4826f23f9e0b" />

However, I have a question about the paper. When you calculate the refraction flow, you use a mask, which already contains the object’s edge information. In that case, is it still necessary to recover the refraction flow?
Also, is the refraction flow not suitable for solid transparent objects with multiple refractions? For solid transparent objects, it seems difficult to perform ray tracing to find the pixel displacement caused by refraction.

**Comment by LJY-XCX** [comment]
Hi Jessica,

The function create_grid_image is used for display images in tensorboard. After the refractive flow with 2 channels is generated, we use the function flow_to_rgb to convert it to a RGB image for visualization. This shouldn't affect the training process.

For your question about mask, in the evaluation process, it's used to avoid the background noise when computing metrices, and in the surface normal prediction pipeline, it's predicted by DeepLabV3 model and used for the depth completion algorithm. I guess your question is about the latter one. In our paper, we did ablation study on different input sources to predict surface normal (including object boundary and refractive flow). In result, we found that the effect of refractive flow is much better than boundary since it contains more geometric informaiton and has a stronger relation to surface normal.

Also, for your final question, we did find that our method fails in scenarios like triangular prism and overlapping objects and we included this limitation in the conclusion section of our paper.

**Comment by jsckdon** [comment]
Hi,Thank you very much for your reply. I wish you all the best in your research and look forward to seeing more of your work in the future.
